### DISCUSSION TITLE
How to install WixSharp on VS Code?

### DISCUSSION THREAD

**Discussion by Fiinleyyy** [oleg-shilo/wixsharp#1550] [Q&A] [answered]
Hi,

excuse me for my incompetence. I'm a computer science student working at a firm and I'm having troubles getting WixSharp to run. 
I find the installation guide(s) fairly confusing. 

Would someone be so nice and help me out with a more detailed installation process? I tried using .NET 8.0 and 6.0 but both don't seem to work for me. When I create a console application running a Hello World is not a problem. But as soon as I download WixSharp nuget package the WixSharp.targets file throws me this error at the "exec" tag:  -532462766

I'm really struggling getting it to work so I would really appreciate some help! :)

And by the way: Which nuget packages do I need to install? WixSharp + WixSharp.bin? And does Wix Toolset come with it?

Yours sincerely Finley

Edit: I'd like to use as many command line inputs if possible since I want to create my Installer through CI/CD later on

**Comment by Fiinleyyy** [comment]
I either get the ms build error code -532462766 or WixSharp library cannot be found

**Comment by oleg-shilo** [comment, marked as answer]
> I find the installation guide(s) fairly confusing.

I would like to address it. So please let me know what areas need improvement or what wiki documentation got you confused.
I have already reviewed and updated the step-by-step tutorial as indeed it needed rework: https://github.com/oleg-shilo/wixsharp/wiki/Building-MSI-%E2%80%93-Step-by-step-tutorial

As for your problem.

There is no WixSharp installation procedure as such as the only tools you need to have in your environment are Visual Studio 2022 and WiX Toolset.

WixSharp is coming into the picture as a WixSharp .NET project template that references WixSharp nuget package. 

Thus you need to:
1. install WiX Toolset `dotnet tool install --global wix`. I recommend v4
2. Create the project from the template and after that you can compile it from either Visual Studio, VSCode or .NET CLI. 

Since only Visual Studio supports templates VSCode can only be used for compilation but not creation of the .NET project.

Thus I have attached for you a sample project that I created from the VS template. You can build it from VSCode, as any other project or from .NET CLI:

```
dotnet build
```
But using Visual Studio is by far a preferred option as it gives you all the answers right out of box: https://github.com/oleg-shilo/wixsharp/wiki/VS2019-%E2%80%93-2022-Templates.

[WixSharp Setup15.zip](https://github.com/user-attachments/files/15752908/WixSharp.Setup15.zip)

**Comment by Fiinleyyy** [comment]
Hi!

Thank you so much for the quick response. I got it to work now and I confused two different errors at the same time, definitely my fault.
Regarding feedback on the documentation I got few things that confused me:

- In the READ.me of the project the installation is already hinted at but only in a few lines while also linking to the documentation [https://github.com/oleg-shilo/wixsharp/wiki]
- The documentation itself also handles how to get started with WixSharp. What confused me here was that there were guides on Wix3 and Wix4 while the command `dotnet tool install --global wix` installed the latest version of Wix, in my case Wix5 which isn't mentioned in the doc. Also the dependencies beneath it confused me: There is a list of dependencies and compilers and extensions where it wasn't clear for me which are actually necessary or not
- Lastly going through this repository lead me to this link [https://github.com/oleg-shilo/wixsharp/wiki/Building-MSI-%E2%80%93-Step-by-step-tutorial] which also handles quick use of getting started with Wixsharp - the many sources led to confusion

Additionally I did not find any mention on which .NET version to use. I assume since the project you linked used version 4.72 that this is the preferred one. But I'm still unsure.

This was my feedback as a thank you for helping me out. Maybe you should update the documentation first after other people mention the same problems since it was only me having problems yet I guess most of the other people were and are able to figure it out.

I have one more detailled question regarding WixSharp but I will ask that in another discussion. Thanks again you may close the thread if you like to!

Greetings

Finley

**Comment by oleg-shilo** [comment]
Thank you Finley. 
I will process your feedback. `Step-by-step-tutorial` is already done (yesterday). But I will review it again.

> I guess most of the other people were and are able to figure it out.

Yes and no 😄
You would not even know how easy to get yourself lost if you only start with the Visual Studio template as probably all others did. 
The template already has everything setup for you: project structure .NET target version, and packages. 
It's not your fault. Not at all. It is just evidence that I need to make it more prominent that VS template is the way to go. And other build alternatives are much trickier.
